### ISSUE TITLE
layouts

### ISSUE THREAD

**Issue by kmcgrath** [SungardAS/condensation#13]
Support the use of a layout so particles can produce parameters, resources, conditions and outputs that always belong together.

Concept:

```
--- layout.template.json.hbs ---
{
  "AWSTemplateFormatVersion": "2010-09-09",
  "Description": {{templateDescription}},
  {{{renderSection 'metadata'}}},
  {{{renderSection 'parameters'}}},
  {{{renderSection 'conditions'}}},
  {{{renderSection 'resources'}}},
  {{{renderSection 'outputs'}}}
}


--- partials/subnets.json.hbs ---

---
subnets:
  -
    name: Subnet1
    default: "10.0.0.0/24"
  -
    name: Subnet2
    default: "10.0.1.0/24"

---
{{#each subnets}}

  {{{section 'parameters'}}}
    {{{partial 'module:particles-common-core' 'parameters_cidr_range' name=this.name default=this.default}}}
  {{{/section}}}

 {{{section 'resources'}}}
    "{{this.name}}Resource": {
      "Type": "AWS::EC2::Subnet",
        "Properties": {
          "CidrBlock": {
            "Ref": "{{this.name}}"
          }
        },
        "VpcId": {
          "Ref": "VpcId"
        }
  {{{/section}}}

  {{{section 'outputs'}}}
    "{{this.name}}": {
      "Value": {
        "Ref": "{{this.name}}Resource"
      }
    },
    "AvailabilityZone": {
      "Value": { "Fn::GetAtt" : [ "{{this.name}}Resource", "AvailabilityZone" ] }
    }
  {{{/section}}}

{{/each}}
```

In this way one partial (new particle type?) could create JSON for all the sections of the template that it needs to touch.

**Comment by kmcgrath** [comment]
Could be great for simple templates, but a pain when adding `DependsOn` and other wait conditions.

**Comment by kmcgrath** [comment]
Implemented in 0.3.0
